### PR TITLE
fix: flow bins xticklabel correction

### PR DESCRIPTION
**Pull request by atownse2**
The issues that this PR fixes are described in #501. This includes:
- Use the correct flow bin index when drawing overflow `xticklabels`
- Draw flow `xticklabel` on overflow bin center instead of using default `xticks`
- Add support for shared axes: Draw overflow bin marker and dashed line on all shared axes, excluding the marker on top of top axis.

Here are some plots I can make with this PR:
5cfac6c0-34c9-41c2-b7b9-05105b80e6d1

9b02ff13-326c-4a9c-bf1d-975d92816e04

### CODE REVIEW

**Comment by atownse2** [comment]
Also, to get the dashed lines and diamonds to draw correctly I needed to draw them on both the top and bottom of each axis. The diamond on the top of the top axis got in the way of labels so I removed it but I left the dashed line because I liked it. I'm not sure if this is the best choice in general.

File: src/mplhep/plot.py [hunk 1]
```diff
@@ -550,68 +550,126 @@ def iterable_not_string(arg):
...
+
+            # Don't draw markers on the top of the top axis
+            top_axis = max(shared_axes, key=lambda a: a.get_position().y0)
+
+            # Draw on all shared axes
+            for _ax in shared_axes:
+                for h in [0, 1]:
+                    _ax.set_xticks(xticks)
+                    _ax.set_xticklabels(xticklabels)
+
```
Comment (andrzejnovak) [inline src/mplhep/plot.py:603]: If you move the skip condition up here, wouldn't that prevent drawing the dashed line on top?

**Comment by andrzejnovak** [comment]
Hey, thanks for the PR and sorry for the delay in reviewing this. This looks mostly good to me, but I am not sure I understand why the dashed line needs to appear on top of the plot as well.

**Comment by atownse2** [comment]
Hi, yes the dashed lines don't need to appear on the top. I thought that it made the overflow bin more clear. What I will do is only include the fixes in this PR and then I will make another PR with some drawing changes that I like.

**Comment by andrzejnovak** [comment]
Thanks, so this now needs to update the reference images. Check https://github.com/scikit-hep/mplhep/blob/master/CONTRIBUTING.md and let me know if you have a problem getting the pytest generation running. N.b. the images generated by pytest might register as "new" in git even if they didn't change, please don't add those to the commit to limit the repo size.

**Comment by andrzejnovak** [comment]
Thanks a lot @atownse2! I'll mint a new release with the fix shortly.

**Pull request merged**